### PR TITLE
Date formatting; add CLI for generating events

### LINKED ISSUE #3: Include any events from today in the "upcoming events" collection
Right now, events held today don't show up as Past Events _or_ as Upcoming Events. This has to do with how those [Collections](https://www.11ty.dev/docs/collections/) are currently being set up in `.eleventy.js`.

**Any events that are being held _today_ should be added to the `upcomingEvents` collection.**

You'll be able to tell they've been added successfully when you run `npm run dev` and navigate to `localhost:8080` — any events held today will show up in the top list.

### PR DESCRIPTION
**Pull request by msutkowski**
Closes #3. 

- Adds a CLI to generate new events
- Adjusts styling on event cards to use flexbox
- Adds date filters for the event card `asDateTime` and `asDateOnly`

### CODE REVIEW

**Comment by BenDMyers** [comment]
Maybe it's just when I ran this and my timezone (~9:20 pm CST on Dec 21), but it seems like the _UI Reproduction Interview Challenge_ item is showing up in both `pastEvents` and `upcomingEvents` somehow. I'm guessing the dates in the events' frontmatter are getting registered in UTC. 🤔 

Screenshot of homepage with two lists. The first list contains UI Reproduction Interview Challenge and Hijacking Screenreaders with CSS. The second list contains Eleventy at 11(pt), Building a Community Calendar with Eleventy, and UI Reproduction Interview Challenge

**Comment by msutkowski** [comment]
@BenDMyers Yeap, that's exactly what's happening. I didn't even think to check that, but `new Date()` on a yyyy-mm-dd is going to initialize at 00 UTC, so this scenario would come up regularly. We could either enforce a standard for using `Number(new Date())`, or we can just do:

```js
  eleventyConfiguration.addCollection('pastEvents', (collectionApi) => {
    return collectionApi
      .getFilteredByGlob('./src/schedule/*.md')
      .filter((event) => isBefore(new Date(event.data.date), new Date()) && !isToday(new Date(event.data.date)));
  });
```

This way it'd only be on one of those lists at a given time. 

The other case I didn't notice was `title: "Michael's Vanity Post"` - In this case, it'd never evaluate as there is no date, so we'd potentially need an 'unscheduled' category if this is the desired behavior.

Thinking about it more, we should probably use a TZ library and enforce a standard so that way we're not worried about copy/tweets/etc conflicting with the server TZ during the build.

**Comment by BenDMyers** [comment]
> In this case, it'd never evaluate as there is no date, so we'd potentially need an 'unscheduled' category if this is the desired behavior.

Yeah, that one's an artifact of our live session from today. I'm thinking once this calendar is up and going, we'd expect every Markdown file to have a date. I'm not sure an unscheduled event is really actionable for anyone visiting. I'm open to thoughts about that, though.

> Thinking about it more, we should probably use a TZ library and enforce a standard so that way we're not worried about copy/tweets/etc conflicting with the server TZ during the build. Or we can just schedule in UTC and not worry about it :)

So, while we didn't get to it today, my thought was that an event's `date` field would eventually contain its start time, as well, so something like

```yaml
date: 2020-12-21 13:00:00
```

That would address the midnight problem we're seeing here, but it forces us to come to a conclusion with standardization. _Some_ timezone likely has to win out here. I think the two main candidates for standardized timezone are **PST** (which is basically the standard we've settled on in the Discord) and **UTC** (global standard).

**Comment by msutkowski** [comment]
@BenDMyers Alright boss, I'm leaving it up to you. You (or the group) let me know what we should do and I'll update this. IMO, UTC would probably be easiest as it's what GH actions run (assuming we use that for the build), but either way, someone has to do a conversion somewhere 🤣. This also brings me back to my favorite point: **time is the worst.**

**Comment by BenDMyers** [comment]
Seems like the noncontroversial take from the community is leaning on UTC!

**Comment by chantastic** [comment]
Hey friends!

Just catching up. Timezones are a real bugger here aren't they!

## Re: invalid dates
My sense is that we should only include valid events in the `pastEvents` helpers.
This wouldn't exclude someone from creating invalid events. They just wouldn't be included in the helper.
This leaves the door open for other event types to be cataloged by type instead of tag.

## Re: UTC
This is an interesting question.
I'm good with chasing UTC for now, given that we don't really have a "headquarters" time zone.
The closest approximation would be PT just because that's the TZ I announce events in. But I'm open to adopting better practices there.

I was looking into GitHub Action `schedule` jobs last night looks like we can rebuild as many times as will fall under the Netlify free tier 😅

**Comment by msutkowski** [comment]
> ## Re: invalid dates
> My sense is that we should only include valid events in the `pastEvents` helpers.
> This wouldn't exclude someone from creating invalid events. They just wouldn't be included in the helper.
> This leaves the door open for other event types to be cataloged by type instead of tag.

What is the definition of a 'valid event'? Title and UTC formatted date?

**Comment by chantastic** [comment]
great question. your definition sounds correct to me!

we'll have optional bits for speakers and technologies. but i don't think
those need to be validated

**Comment by msutkowski** [comment]
@chantastic @BenDMyers Sorry about the delay on updating this. 

This should behave like expected at this point. I've added a little CLI to help generate the md files _(note: it intentionally doesn't cover every edge case that could come up, but hopefully it's a starting point for someone else to have some fun 🤣  )_

Just run:
```sh
yarn gen
```

**Comment by BenDMyers** [comment]
The new CLI is _very_ handy - thank you @msutkowski!

File: src/_includes/layouts/homepage.html [hunk 1]
```diff
@@ -23,6 +26,7 @@ <h2 id="upcoming-events-section">Upcoming Events</h2>
           {% if event.data.type %}
           <span>{{ event.data.type }}</span>
           {% endif %}
+          <p>{{ event.data.date }}</p>
```
Comment (BenDMyers) [inline src/_includes/layouts/homepage.html:29]: Thank you so much for adding the date to the homepage - this will seriously help a lot! 😅

**Comment by msutkowski** [comment]
> The new CLI is _very_ handy - thank you @msutkowski!

:tada: I think this could just make for good ol' fashion programming fun as there are so many little puzzles in here. We could make this async, where the choices for speakers are returned from different sources as well as take custom inputs. Could also load seed data based on event types, etc etc. So many options! Maybe this could spin-off it's own fun lunch.dev segment --- I've left misc. comments with improvement ideas in the code for future contributors.

The only thing I'm really concerned about here is the TZ stuff. This should work fine, but as always, dates are the worst :)

**Review by BenDMyers — APPROVED** [review]

**Review by chantastic — APPROVED** [review]
I WANT IT ALL!

File: utils/event-types.js [hunk 2]
```diff
@@ -0,0 +1 @@
+module.exports = ['Talk', 'Episode Party'];
```
Comment (BenDMyers) [inline utils/event-types.js:1]: We should add `'Mob Job'` as a type, too.

**Comment by BenDMyers** [comment]
@msutkowski, this is great work. Would you be willing to pull down the latest changes in `main`, resolve merge conflicts, and get this PR merged in?

**Comment by BenDMyers** [comment]
@all-contributors please add @msutkowski for code and tooling

**Pull request merged**